iib/workers/tasks/build.py

```diff
@@ -439,34 +448,32 @@ def _push_image(request_id, arch):
     :param str arch: the architecture of the container image to push
     :raises IIBError: if the push fails
     """
-    source = _get_local_pull_spec(request_id, arch)
+    source = _get_local_pull_spec(request_id, arch, include_transport=True)
     destination = _get_external_arch_pull_spec(request_id, arch, include_transport=True)
     log.info('Pushing the container image %s to %s', source, destination)
-    run_cmd(
-        ['podman', 'push', '-q', source, destination],
+    # Use "skopeo copy" because "podman push" doesn't support setting the format
+    _skopeo_copy(
+        source,
+        destination,
         exc_msg=f'Failed to push the container image to {destination} for the arch {arch}',
     )
 
-    log.debug(f'Verifying that {destination} was pushed as a v2 manifest due to RHBZ#1810768')
+    log.debug(f'Verifying that {destination} was pushed as a v2 manifest')
     skopeo_raw = skopeo_inspect(destination, '--raw')
     if skopeo_raw['schemaVersion'] != 2:
-        log.warning(
-            'The manifest for %s ended up using schema version 1 due to RHBZ#1810768. Manually '
-            'fixing it with skopeo.',
-            destination,
-        )
-        exc_msg = f'Failed to fix the manifest schema version on {destination}'
-        _skopeo_copy(destination, destination, exc_msg=exc_msg)
+        raise IIBError(f'Manifest at {destination} is not schema version 2')
 
 
 @retry(wait_on=IIBError, logger=log)
-def _skopeo_copy(source, destination, copy_all=False, exc_msg=None):
+def _skopeo_copy(source, destination, copy_all=False, dest_token=None, exc_msg=None):
     """
     Wrap the ``skopeo copy`` command.
 
     :param str source: the source to copy
     :param str destination: the destination to copy the source to
     :param bool copy_all: if True, it passes ``--all`` to the command
+    :param str dest_token: the token to pass to the ``--dest-token` parameter of the command.
+        If not provided, ``--dest-token`` parameter is also not provided.
```

Comment:
Should this be ``--dest-token`` parameter is also not provided. ==> ``--dest-authfile`` parameter is also not provided.